deltacat/io/ray_plasma_object_store.py

```diff
@@ -23,6 +29,25 @@ def get_many(self, refs: List[Any], *args, **kwargs) -> List[object]:
         loaded_refs = [cloudpickle.loads(obj_id) for obj_id in refs]
         return ray.get(loaded_refs)
 
+    def delete_many(self, refs: List[Any], *args, **kwargs) -> List[object]:
+        object_refs = [cloudpickle.loads(obj_id) for obj_id in refs]
+        all_deleted = True
+        start = time.monotonic()
+        try:
+            # Per Ray docs, deletion success will NOT be returned. In-use objects will be deleted when ref count is 0.
+            # https://github.com/ray-project/ray/blob/master/python/ray/_private/internal_api.py
+            free(object_refs)
```

Comment:
If free doesn't work, how do we delete it from the object store?

Reply:
> The object will be automatically deleted when the ref counts = 0. So, we can call del in python.

Reply:
`free()` isn't doing nothing though, right? The documentation says it will still send some signal to the object store, and once those ref counts go to 0, then it will be removed from the object store. I don't see where it says the object will be automatically deleted *without* calling `free()`

Reply:
Sure... should I just remove the `delete_many` implementation from ray plasma object store then? If this is true, then we don't need it

Reply:
yeah.. throw error maybe. Ray plasma doesn't support delete method.

Reply:
I don't think we should throw error, since we're calling `delete_many` regardless of underlying object store.

Reply:
You can unpin by deleting the ref actually.

Reply:
We already automatically delete every object ref immediately after we call put, but we also store a ref in the hash bucket results, which I believe wouldn't go away until the round ends. So my understanding is that if we use the ray plasma store, it should automatically clean up the object store without us doing anything?

Reply:
Correct.